lib/Dialect/SV/Transforms/PrettifyVerilog.cpp

```diff
@@ -51,11 +65,83 @@ struct PrettifyVerilogPass
   bool splitAssignment(OpBuilder &builder, Value dst, Value src);
 
   bool anythingChanged;
+  SideEffectTracker *tracker;
 
   DenseSet<Operation *> toDelete;
 };
 } // end anonymous namespace
 
+SideEffectTracker::SideEffectTracker(hw::HWModuleOp module) {
+  // Visit all side-effecting operations in procedural regions.
+  module.walk([&](Operation *op) {
```

Comment:
A blind walk for all operations is going to add several seconds to runtime.  Since nesting is pretty shallow and blocking assigns relatively rare, perhaps compute safety on-demand?  a procedural region isn't tainted by safety concerns until you see the first bpassign, you only need to keep track of expressions who are derived from a value used on the lhs of a blocking assign, etc.